File: ios/AnnaNative.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(AnnaNative, NSObject)

RCT_EXTERN_METHOD(showToast:(NSString *)text duration:(NSTimeInterval)duration)

RCT_EXTERN_METHOD(showLoading:(NSString *)text)

RCT_EXTERN_METHOD(hideLoading)

@end

File: ios/AnnaNative.swift
@objc(AnnaNative)
public class AnnaNative: NSObject {
    
    @objc(showToast:duration:)
    public func showToast(_ text: String, duration: TimeInterval) -> Void {
        DispatchQueue.main.async {
            Toast.show(text: text, duration: duration);
        }
    }
    
    @objc(showLoading:)
    public func showLoading(_ text: String) -> Void {
        DispatchQueue.main.async {
            Toast.showLoading(text: text);
        }
    }
    
    @objc(hideLoading)
    public func hideLoading() -> Void {
        DispatchQueue.main.async {
            Toast.hideLoading();
        }
    }
}
